src/main/java/com/spotinst/sdkjava/model/SpotManagedInstanceClient.java

```diff
@@ -0,0 +1,211 @@
+package com.spotinst.sdkjava.model;
+
+import com.spotinst.sdkjava.exception.HttpError;
+import com.spotinst.sdkjava.exception.SpotinstHttpException;
+import com.spotinst.sdkjava.model.bl.aws.managed.instance.ManagedInstance;
+import com.spotinst.sdkjava.model.requests.aws.managed.instance.AwsManagedInstanceRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class SpotManagedInstanceClient {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(SpotManagedInstanceClient.class);
+    //Members
+    private              String                      authToken;
+    private              String                      account;
+    private              ISpotManagedInstanceRepo    spotManagedInstanceRepo;
+
+    //Constructor
+    public SpotManagedInstanceClient(String authToken, String account) {
+        this.authToken = authToken;
+        this.account = account;
+
+        setSpotinstManagedInstanceRepo();
+
+    }
+
+    public ISpotManagedInstanceRepo getSpotManagedInstanceRepo() {
+        return spotManagedInstanceRepo;
+    }
+
+    public void setSpotinstManagedInstanceRepo() {
+        this.spotManagedInstanceRepo =
+                SpotinstRepoManager.getInstance().getSpotManagedInstanceRepo();
+    }
+
+    //Methods
+    public ManagedInstance createManagedInstance(AwsManagedInstanceRequest managedInstanceCreationRequest) {
+        ManagedInstance retVal;
+
+        ManagedInstance managedInstanceToCreate = managedInstanceCreationRequest.getManagedInstance();
+
+        RepoGenericResponse<ManagedInstance> creationResponse =
+                getSpotManagedInstanceRepo().create(managedInstanceToCreate, authToken, account);
+
+        if (creationResponse.isRequestSucceed()) {
+            retVal = creationResponse.getValue();
+        }
+        else {
+            List<HttpError> httpExceptions = creationResponse.getHttpExceptions();
+            HttpError       httpException  = httpExceptions.get(0);
+            LOGGER.error(String.format(
+                    "Error encountered while attempting to create AWS Managed Instance. Code: %s. Message: %s.",
+                    httpException.getCode(), httpException.getMessage()));
+            throw new SpotinstHttpException(httpException.getMessage());
+        }
+
+        return retVal;
+    }
+
+    public Boolean updateManagedInstance(AwsManagedInstanceRequest managedInstanceCreationRequest, String managedInstanceId) {
```

Comment:
The input parameter name should be managedInstanceUpdateRequest

Reply:
changed